src/test/java/tdl/record_upload/RecordAndUploadAppTest.java

```diff
@@ -52,6 +52,31 @@ void sendStop() throws Exception {
 
     @Test
     public void orchestratesMultipleThreads() throws Exception {
+        runRecordAndUploadApp();
```

Comment:
The assertions do not make sense unless you understand what this method does. 
On that basis, it is better to have it inline. It is a fairly involved test and that needs to be read carefully.

Reply:
Fixed